### PR TITLE
Updated lark.fg and test cases to support basic functionalites

### PR DESCRIPTION
**Pull request by nisargshah2000**

### CODE REVIEW

File: parser/fg.lark [hunk 1]
```diff
@@ -1,51 +1,83 @@
...
 
-points : 		"[" NUMBER ("," NUMBER)* "]"
+plotting_stmt : dependent_variable "<-" "func" indendent_variable [attr_list]
 
-subplot_stmt : 		"subplot" INT ("," INT)? "{" stmt_list "}"
+dependent_variable : ID
+indendent_variable : ID
 
-plotting_stmt : 	IDNAME "func" IDNAME
+function_stmt : dependent_variable "<-" (single_function | function_list)
```
Comment (vedantsingh1404) [inline parser/fg.lark:57]: should be independent variable instead.

File: parser/fg.lark [hunk 2]
```diff
@@ -1,51 +1,83 @@
...
 
-func 	:		"func" //filler for testing purposes
+function_list : "[" function ("," function)* "]"
+
+FUNCTION : (LCASE_LETTER|REAL|PAREN|OPERATOR)+
+
+PAREN : "(" | ")"
+OPERATOR : "+" | "-" | "*" | "/" 
+
+ID : (LCASE_LETTER)+
```
Comment (vedantsingh1404) [inline parser/fg.lark:70]: Why can't the figure name consist of upper case letters?

File: parser/fg.lark [hunk 3]
```diff
@@ -1,51 +1,83 @@
...
+       | plotting_stmt 
+       | range_stmt
+       | function_stmt
+
+size_stmt : "size <-" nrows ["," ncols] [attr_list]
+
+nrows : INT
+ncols : INT 
+
+range_stmt : variable "<-" (points | range)
```
Comment (vedantsingh1404) [inline parser/fg.lark:26]: why not use independent variable instead?

File: parser/fg.lark [hunk 4]
```diff
@@ -1,51 +1,83 @@
...
 
-range_descriptor:	INT "in" 
-			| "continuous" 
 
-points : 		"[" NUMBER ("," NUMBER)* "]"
+plotting_stmt : dependent_variable "<-" "func" indendent_variable [attr_list]
 
-subplot_stmt : 		"subplot" INT ("," INT)? "{" stmt_list "}"
+dependent_variable : ID
+indendent_variable : ID
```
Comment (vedantsingh1404) [inline parser/fg.lark:55]: typo x 2

**Review by vedantsingh1404 — CHANGES_REQUESTED** [review]

File: parser/fg.lark (hunk 2 above)
Comment (nisargshah2000) [reply parser/fg.lark:70]: Just for simplicity, all variable and ID are lower case.

File: parser/fg.lark (hunk 3 above)
Comment (nisargshah2000) [reply parser/fg.lark:26]: We can use that also, but it is not used just for readability. In general, dependent and independent variable comes together.

File: parser/fg.lark (hunk 4 above)
Comment (nisargshah2000) [reply parser/fg.lark:55]: I will correct it in next commit.

File: parser/fg.lark (hunk 2 above)
Comment (vedantsingh1404) [reply parser/fg.lark:70]: @rajp152k should we add support for upper case variable names also? Won't make much difference though.

File: parser/fg.lark (hunk 3 above)
Comment (vedantsingh1404) [reply parser/fg.lark]: I agree, but I guess since this variable here is the independent one, why not reduce the clutter.

**Comment by vedantsingh1404** [comment]
Seems fine, merging this for now. The upper case changes can be made later as well.

**Review by vedantsingh1404 — APPROVED** [review]

File: parser/fg.lark (hunk 3 above)
Comment (nisargshah2000) [reply parser/fg.lark]: Changes done.

File: parser/fg.lark (hunk 2 above)
Comment (rajp152k) [reply parser/fg.lark:68]: yeah, a matter of taste... 
maybe just enforce the first letter to be small and let the rest be alphanumeric + underscore for identifiers.
not of high priority right now.

**Review by rajp152k — COMMENTED** [review]
reviewed; 
basics all okay.
probably a few additions and changes in the future for 3d plots(eg:- what's dependent and independent will depend on the equations themselves)

**Pull request merged**